pep-0432.txt

```diff
@@ -662,47 +668,25 @@ Python data types (not set == ``NULL``) or numeric flags (not set == ``-1``)::
          * be set before calling PyRun_PrepareMain (Py_ReadMainInterpreterConfig
          * will set one of them based on the command line arguments if
          * prepare_main is non-zero when that API is called).
-        int prepare_main;
+        PyBoolObject    *prepare_main;
         PyUnicodeObject *main_source; /* -c switch */
         PyUnicodeObject *main_path;   /* filesystem path */
         PyUnicodeObject *main_module; /* -m switch */
         PyCodeObject    *main_code;   /* Run directly from a code object */
         PyObject        *main_stream; /* Run from stream */
-        int run_implicit_code;        /* Run implicit code during prep */
+        PyBoolObject    *run_implicit_code; /* Run implicit code during prep */
 
         /* Interactive main
          *
          * Note: Settings related to interactive mode are very much in flux.
          */
         PyObject *prompt_stream;      /* Output interactive prompt */
-        int show_banner;              /* -q switch (inverted) */
-        int inspect_main;             /* -i switch, PYTHONINSPECT */
+        PyBoolObject *show_banner;    /* -q switch (inverted) */
+        PyBoolObject *inspect_main;   /* -i switch, PYTHONINSPECT */
 
     } PyMainInterpreterConfig;
 
-
-    /* Struct initialization is pretty horrible in C89. Avoiding this mess would
-     * be the most attractive aspect of using a PyDictObject* instead... */
-    #define _PyArgConfig_INIT  NULL, NULL, NULL, NULL
```

Comment:
Ah, these can all go away because you switched to pointers and presumably using a C99 compiler will automatically initialize PyMainInterpreterConfig to all NULLs.